task/packrat/src/gimlet.rs

```diff
@@ -117,12 +117,33 @@ impl GimletData {
     }
 
     pub(crate) fn get_spd_data(
-        &mut self,
+        &self,
         index: usize,
     ) -> Result<u8, RequestError<Infallible>> {
         self.spd_data
             .get(index)
             .copied()
             .ok_or(RequestError::Fail(ClientError::BadMessageContents))
     }
+
+    pub(crate) fn get_full_spd_data(
+        &self,
+        dev: usize,
+        out: LenLimit<Leased<idol_runtime::W, [u8]>, 512>,
+    ) -> Result<(), RequestError<Infallible>> {
+        if out.len() != spd::MAX_SIZE {
```

Comment:
Could `out.len()` be safely greater than `spd::MAX_SIZE` ?

Reply:
That would be technically safe, but would suggest an error on the part of the callers; it would be weird to copy data into a subsection of the lease.